src/poliastro/twobody/orbit.py

```diff
@@ -281,7 +281,7 @@ def propagate(self, value, method=mean_motion, rtol=1e-10, **kwargs):
 
             return propagate(self, time_of_flight, method=method, rtol=rtol, **kwargs)
 
-    def sample(self, values=None, method=mean_motion):
+    def sample(self, values=None, method=mean_motion, dense_output=False):
```

Comment:
Instead of exposing `dense_output` to the user, I propose doing something else, since this is a very low level detail and can lead to confusion. For instance, they could do

```
sample(..., method=kepler, dense_output=True)
```

and it's not clear what should happen in this case. We can either:

* Make `dense_output=True/False` a property of the propagator. The simplest, although a bit ugly way, would be doing something like

```
def cowell(...):
    ...

cowell.dense_output = True
kepler.dense_output = False
```

we could also use some dictionary somewhere:

```
DENSE_OUTPUT_METHODS = [cowell]

...
if method in DENSE_OUTPUT_METHODS:
    ...
```

or turn the propagators into classes:

```
class Cowell(Propagator):
    dense_output = True
    def __call__(self, ...):
        ...

...
orbit.sample(..., method=Cowell())
```

but I find this last option the least appealing if we're going this way.

* Move the propagation loop to the propagator code, instead of having it in `Orbit._sample`. This could also be done in several ways: in particular, `cowell` already accepts a list of values in `tof` (something that the other propagators don't) so perhaps we should generalize this behavior and, if `kepler` gets a list of `tof`, do the for loop inside.